proxy/coco/src/peer/run_state.rs

```diff
@@ -140,9 +158,10 @@ pub struct RunState {
     /// `Connected(Peer1) -> Connected(Peer1) -> Disconnecting(Peer1)`
     //
     // FIXME(xla): Use a `Option<NonEmpty>` here to express the invariance.
-    connected_peers: HashMap<PeerId, usize>,
```

Comment:
We no longer get messages about `Connected` and `Disconnected` so we don't need this `HashMap`. Instead, we can poll for the set of connected peers